### PR TITLE
Add filter by owner and team to issue/pulls search endpoint

### PR DESCRIPTION
**Pull request by jpraet**

### CODE REVIEW

File: routers/api/v1/repo/issue.go [hunk 1]
```diff
@@ -130,6 +138,37 @@ func SearchIssues(ctx *context.APIContext) {
...
+				ctx.Error(http.StatusInternalServerError, "GetUserByName", err)
+			}
+			return
+		}
+		opts.OwnerID = owner.ID
+		opts.AllLimited = false
+		opts.AllPublic = false
+		opts.Collaborate = util.OptionalBoolFalse
+	}
+	if ctx.Form("team") != "" {
```
Comment (lunny) [inline routers/api/v1/repo/issue.go]: So maybe move this to line 155 is better, then we could delete line 157-160

Comment (jpraet) [reply routers/api/v1/repo/issue.go]: But then we can't return an error message to the user when the operation is called with missing "owner" parameter when filtering on team? The "team" parameter would be silently ignored.

**Review by lunny — APPROVED** [review]

File: routers/api/v1/repo/issue.go [hunk 2]
```diff
@@ -130,6 +138,37 @@ func SearchIssues(ctx *context.APIContext) {
...
+		if err != nil {
+			if models.IsErrUserNotExist(err) {
+				ctx.Error(http.StatusBadRequest, "Owner not found", err)
+			} else {
+				ctx.Error(http.StatusInternalServerError, "GetUserByName", err)
+			}
+			return
+		}
+		opts.OwnerID = owner.ID
+		opts.AllLimited = false
```
Comment (6543) [inline routers/api/v1/repo/issue.go:152]: why does owner affect AllLimited or AllPublic?

and tests should be extend

Comment (jpraet) [reply routers/api/v1/repo/issue.go:152]:
```
	AllPublic       bool // Include also all public repositories of users and public organisations
	AllLimited      bool // Include also all public repositories of limited organisations
```

If those are set to true it returns those repo's, regardless of whether the owner matches or not.

I'll extend the `api_issue_test.go`

**Review by 6543 — APPROVED** [review]

**Pull request merged**